pkg/nodelist/impl.go

```diff
@@ -25,26 +27,52 @@ func (n *nodeList) Start(nodeID model.NodeID, publicAddr string) {
 		n.publicAddr = publicAddr
 		log.Log().Infof("Registering local node on nodelist (id=%s,addr=%s)", nodeID, publicAddr)
 		data, _ := json.Marshal(model.NodeInfo{ID: nodeID, Address: publicAddr})
-		n.documents.AddDocument(&model.Document{Timestamp: time.Now(), Contents: data, Type: nodeInfoDocumentType})
+		document := model.Document{
+			Timestamp: time.Now(),
+			Type:      nodeInfoDocumentType,
+		}
+		document.Hash = model.CalculateDocumentHash(document.Type, document.Timestamp, data)
+		if err := n.documents.AddDocumentWithContents(document, bytes.NewReader(data)); err != nil {
```

Comment:
These 2 lines would be 1 if `AddDocumentWIthContents` handled the timestamp and hash

Reply:
True, also that's a good practice when there aren't too many parameters ("accept the smallest and least coupled datastructures as possible" or something like that). I'll change it.